main/src/main/java/org/mobilitydata/gtfsvalidator/validator/TripAndShapeDistanceValidator.java

```diff
@@ -53,38 +51,33 @@ public class TripAndShapeDistanceValidator extends FileValidator {
 
   @Override
   public void validate(NoticeContainer noticeContainer) {
-    List<String> uniqueShapeIds =
-        shapeTable.getEntities().stream()
-            .map(GtfsShape::shapeId)
-            .distinct()
-            .collect(Collectors.toList());
-
-    uniqueShapeIds.forEach(
-        shapeId -> {
-          double maxShapeDist =
-              shapeTable.getEntities().stream()
-                  .filter(s -> s.shapeId().equals(shapeId))
-                  .mapToDouble(GtfsShape::shapeDistTraveled)
-                  .max()
-                  .orElse(Double.NEGATIVE_INFINITY);
-
-          tripTable
-              .byShapeId(shapeId)
-              .forEach(
-                  trip -> {
-                    double maxStopTimeDist =
-                        stopTimeTable.byTripId(trip.tripId()).stream()
-                            .mapToDouble(GtfsStopTime::shapeDistTraveled)
-                            .max()
-                            .orElse(Double.NEGATIVE_INFINITY);
-
-                    if (maxStopTimeDist > maxShapeDist) {
-                      noticeContainer.addValidationNotice(
-                          new TripDistanceExceedsShapeDistanceNotice(
-                              trip.tripId(), shapeId, maxStopTimeDist, maxShapeDist));
-                    }
-                  });
-        });
+    shapeTable
+        .byShapeIdMap()
+        .forEach(
+            (shapeId, shape) -> {
+              double maxShapeDist =
+                  shapeTable.byShapeId(shapeId).stream()
```

Comment:
This isn't super inefficient, but it seems weird to look up the shape point here when you already have them specified in the `shape` variable above.

Reply:
From my understanding of [the specifications](https://gtfs.org/schedule/reference/#shapestxt), in `shapes.txt`, the `shape_id` is not a unique id. 

> ID - An ID field value is an internal ID, not intended to be shown to riders, and is a sequence of any UTF-8 characters. Using only printable ASCII characters is recommended. **An ID is labeled "unique ID" when it must be unique within a file.** IDs defined in one .txt file are often referenced in another .txt file. IDs that reference an ID in another table are labeled "foreign ID".

So what I'm doing here is trying to group by `shape_id` to find the maximum value of the point. I'm all ears if you can think of any other way that would be more efficient.